openlibrary/plugins/upstream/mybooks.py

```diff
@@ -193,7 +193,12 @@ def render(self, page=1, sort='desc', list=None):
             self.counts['sponsorships'] = len(sponsorships)
 
             if self.key == 'sponsorships':
-                data = self._prepare_data(logged_in_user, sponsorships=sponsorships)
+                data = add_availability(
+                    web.ctx.site.get_many([
+                        '/books/%s' % doc['openlibrary_edition']
+                        for doc in sponsorships
+                    ])
+                )
```

Comment:
```suggestion
                ) if sponsorships else None
```
We redirect to `/people/{username}` when `data == None` at the end of this method.

If we send an empty list when a patron has no sponsorships, they can still navigate to the sponsorship page using their browser's address bar.